ENIGMAsystem/SHELL/Graphics_Systems/Direct3D11/DX11textures.cpp

```diff
@@ -65,8 +59,16 @@ int graphics_create_texture(const RawImage& img, bool mipmap, unsigned* fullwidt
   tdesc.CPUAccessFlags = 0;
   tdesc.MiscFlags = 0;
 
-  if (FAILED(m_device->CreateTexture2D(&tdesc,&tbsd,&tex)))
-    return 0;
+  if (img.pxdata != nullptr && (img.w != *fullwidth || img.h != *fullheight)) {
+    RawImage padded = image_pad(img, *fullwidth, *fullheight);
```

Comment:
Honestly, why even have image_pad accept a const& and return a new RawImage, rather than just making it a method on RawImage?

Reply:
Good question for fundies, we can talk about that later.

Reply:
Also, mutability has its own performance caveats, although I'm not sure how harmful they are in this use case. Regardless, the contract did state the original image is preserved. So that change would necessitate an update of the postconditions.
https://github.com/enigma-dev/enigma-dev/blob/560ad579a8dcb9ab23f5a521274d518fef574cd6/ENIGMAsystem/SHELL/Graphics_Systems/graphics_mandatory.h#L44